spec/api/keys/keys_spec.rb

```diff
@@ -656,17 +563,262 @@
+      "username" => org_admin_name,
+      "email" => "#{org_admin_name}@#{org_admin_name}.com"
+    }
+    end
+    let (:org_user_payload) do
+      {
+        "public_key" => keys[:org_user][:public],
+        "username" => org_user_name,
+        "email" => "#{org_user_name}@#{org_user_name}.com"
+      }
+    end
+    let (:other_org_user_payload) do
+      {
+        "public_key" => keys[:other_org_user][:public],
+        "username" => other_org_user_name,
+        "email" => "#{other_org_user_name}@#{other_org_user_name}.com"
+      }
+    end
+    let (:org_client_payload) do
+      {
+        "public_key" => keys[:org_client][:public],
+        "name" => org_client_name,
+        "admin" =>false
+      }
+    end
+
+    let (:other_org_client_payload) do
+      {
+        "public_key" => keys[:other_org_client][:public],
+        "name" => other_org_client_name,
+        "admin" =>false
+      }
+    end
+    let (:unassociated_requestor) do
+      requestor(user['name'], user['private_key'])
+    end
+
+    before :all do
+      # Create other org. This org has no members other than those created automatically
+
+
+      post("#{platform.server}/organizations", superuser, :payload => $other_org_payload).should look_like({:status => 201})
+    end
+    after :all do
+      delete("#{platform.server}/organizations/#{$other_org_name}/clients/#{$other_org_name}-validator", superuser).should look_like({:status => 200})
+      delete("#{platform.server}/organizations/#{$other_org_name}", superuser).should look_like({:status => 200})
+    end
+
+    before :each do
+      # status checks here to provide a level of protection against having to chase down unexpected errors
+      # because our startup state isn't what we think it is.
+      post("#{platform.server}/users", superuser, :payload => base_user_payload.dup.merge(org_admin_payload)).should look_like({:status => 201} )
+      post("#{platform.server}/users", superuser, :payload => base_user_payload.dup.merge(org_user_payload)).should look_like({:status => 201} )
+      post("#{platform.server}/users", superuser, :payload => base_user_payload.dup.merge(other_org_user_payload)).should look_like({:status => 201} )
+
+      post("#{org_base_url}/clients", superuser, :payload => org_client_payload).should look_like({:status => 201})
+      post("#{org_base_url}/users", superuser, :payload => { "username" => org_user_name} ).should look_like({:status => 201})
+      post("#{org_base_url}/users", superuser, :payload => { "username" => org_admin_name} ).should look_like({:status => 201})
+
+      platform.add_user_to_group($org['name'], org_admin, "admins")
+
+      post("#{platform.server}/organizations/#{$other_org_name}/clients", superuser, :payload => other_org_client_payload).should look_like({:status => 201})
+      post("#{platform.server}/organizations/#{$other_org_name}/users", superuser, :payload => { "username" => other_org_user_name } ).should look_like({:status => 201})
     end
```

Comment:
Many of them are, at least at the context-level. I'll take a pass to see if I can thin it down a bit.  

However, there's not a lot of value in having this done for 'each' either - given what we're testing in these new tests (the behavior of retrieving keys) setting the environment to a known state once, and tearing it down once across the suite/context would be cleaner. And faster to execute.  And a pain in the ass because rspec is opinionated ;)